### ISSUE TITLE
Warn the user of potential duplicates

### ISSUE THREAD

**Issue by ZeHiro** [bnjbvr/kresus#395]
It appears when there is a difference in the real balance (ie the one on the bank website) and the displayed one by kresus is a real problem for some users, and prevents them to use kresus day to day.
This often happens when some duplicates are imported by Kresus. 

A simple way to manage this would be to warn the user of potential duplicates, so that s.he can fix this manually. 

Here is my solution proposal : 
- On fetching new operation, also fetch account balance from weboob
- No need to save it in the DB as new operations/balance is always refetched on startup of kresus (so just a memory save is enough, map with a key which is the account DB id could make it)
- In the /all/ request, include the lastFetchedBalance for each accound
- Add a warning when the client detected a difference between lastFetchedBalance and sum(operations)+initialAmount

**Comment by ArchangeGabriel** [comment]
Good idea, but it might also be because Kresus did NOT import two similar operations. I haven’t any duplicate, but I have three operations missing, because each of them had another operation of same amount/date/“merchant”.

**Comment by bnjbvr** [comment]
Thank you @ZeHiro for the idea, I like it and have thought about using the bank's balance for mapping existing transactions with bank ones. Although there is just one issue I've been told about in the weboob IRC channel: some banks differ the computation of the balance and the arrival of new transactions. That is, if today you have 10 euros on your bank account, and tomorrow you get a new transaction of -5 euros, there can be a period during which balance will say "10 euros" although you really only have 5 euros; the balance is recomputed later.

I think we still need to experiment with this, though.

@ArchangeGabriel, this should be fixed in the next version of Kresus, thanks to @ZeHiro 's work: https://github.com/bnjbvr/kresus/commit/d30f948b0486d03040b09bd9eccf42916389afa2

**Comment by ZeHiro** [comment]
@ArchangeGabriel 

Next version will partially correct this (operations with the same sate/merchant/amount present at the same time on the website will be imported). There will still be some edge cases where operations will not be imported(but I think they will now be very rare).  You can correct this by adding manually the operation, in the settings part > click on the '+'

**Comment by ArchangeGabriel** [comment]
@bnjbvr @ZeHiro OK, nice. I know about adding operation manually, but it doesn’t give the same results.

@bnjbvr Indeed, you’re right. For instance, my bank take into account some future operations (scheduled transfers…) for the balance. But they also give a « current balance » that is not taking into account such things, rather just operations already done. But sometimes (around update times, so 23h-00h), operations appear in both categories, and one or two of the balances are wrong.

**Comment by ZeHiro** [comment]
@ArchangeGabriel what do you mean by "it doesn't give the same result"?

**Comment by ArchangeGabriel** [comment]
@ZeHiro I don’t have access to my instance currently, but from what I remember, it was something with operation label, things like imported operation have a different displayed label and detailed label but created ones can only have the same thing in both fields.

**Comment by ZeHiro** [comment]
Yes you're right, it is a design choice, as it is manually added, I supposed it wasn't necessary to make the difference.

**Comment by ArchangeGabriel** [comment]
Well, for my bank at least, they have different meaning and contains different information, and both are useful. That said, I understand your design choice, and I can live with it as long as my operations are correctly imported, which should be the case starting next version. BTW, will old missed operations be imported automatically?

**Comment by ZeHiro** [comment]
They won't, for two reasons : 
- first the operations could not be visible on you're bank website (for exemple, LCL keeps the operations 45days on the website)
- if a similar operation is already in the DB it won't be imported (otherwise you would have 100s of duplicates), and there is no way to make the distinction for kresus. Only the new cases of similar operations imported at the same time will be correctly imported.

**Comment by ArchangeGabriel** [comment]
I know about the first point, and that is for me the main reason to use something like kresus. In my case (LBP), they are 4 visible months, and the concerned operations are less than two months old (happened for the first time around https://github.com/bnjbvr/kresus/issues/373).

For the second point, will it work if I remove the already imported one manually using Cozy’s data-inspector?

**Comment by ZeHiro** [comment]
I think it will, could you confirm @bnjbvr ?

**Comment by bnjbvr** [comment]
Yes, this workaround should work.

**Comment by ArchangeGabriel** [comment]
OK, thanks. :)
